src/go/configinfo/service_info.go

```diff
@@ -716,9 +716,12 @@ func (s *ServiceInfo) processTypes() error {
 			}
 		}
 
-		// Store the mapping per selector.
+		// Replace the snake name with the json name in url template
 		if len(snakeToJson) > 0 {
-			mi.SegmentMappings = snakeToJson
+			for _, httpRule := range mi.HttpRule {
```

Comment:
This is the wrong place to do it. This can lead to problems in route matching. We shouldn't modify the envoy route config with JSON name, only the backend routing filter.

- Example: /v1/{book=books_segment/*}
- This request should match: /v1/books_segment/42
- But you'll change the UriTemplate to /v1/{book=booksSegment/*}
- Now that request won't match.

Reply:
We only change filed_name, not value.   In above case, we will not replace books_segment which is not a field name.

Reply:
Ack got it, was confused. Then this is fine, i don't see any other downsides to changing it here.